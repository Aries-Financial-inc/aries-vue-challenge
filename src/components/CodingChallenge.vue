<template>
  <div>
    <h1>Options Profit Calculator</h1>
    <div class="d-flex justify-content-center">
      <risk-reward-chart :options="optionsData"/>
    </div>
  </div>
</template>

<script>
import RiskRewardChart from "./RiskRewardChart"
export default {
  name: 'CodingChallenge',
  props: {
    optionsData: Array
  },
  components: {
    RiskRewardChart,
  }
}
</script>

<style scoped>
.d-flex{
  display: flex;
}

.justify-content-center{
  justify-content: center;
}
</style>
