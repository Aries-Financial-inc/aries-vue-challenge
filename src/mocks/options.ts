export const sampleData = [
  {
    strike_price: 100,
    type: 'Call',
    bid: 10.05,
    ask: 12.04,
    position: 'Long',
    expiration_date: '2025-12-17T00:00:00Z'
  },
  {
    strike_price: 102.5,
    type: 'Call',
    bid: 12.1,
    ask: 14,
    position: 'Long',
    expiration_date: '2025-12-17T00:00:00Z'
  },
  {
    strike_price: 103,
    type: 'Put',
    bid: 14,
    ask: 15.5,
    position: 'Short',
    expiration_date: '2025-12-17T00:00:00Z'
  },
  {
    strike_price: 105,
    type: 'Put',
    bid: 16,
    ask: 18,
    position: 'Long',
    expiration_date: '2025-12-17T00:00:00Z'
  }
]
