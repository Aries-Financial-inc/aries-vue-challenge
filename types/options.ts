export interface OptionContract {
  type: "call" | "put";
  strikePrice: number;
  premium: number;
  quantity: number;
}

export interface AnalysisResult {
  priceAtExpiry: number;
  profitLoss: number;
}

export interface CombinedAnalysisResult {
  priceAtExpiry: number;
  [key: string]: number; // Index signature for dynamic keys
}

export interface OptionDetails {
  type: "call" | "put";
  strikePrice: number;
  longShort: "long" | "short";
  maxProfit: number | string;
  maxLoss: number | string;
  breakeven: number;
}
