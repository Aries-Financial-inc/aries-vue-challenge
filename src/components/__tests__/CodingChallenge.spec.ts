import { shallowMount } from '@vue/test-utils'
import CodingChallenge from '@/components/CodingChallenge.vue'
import { sampleData } from '@/mocks/options'

describe('CodingChallenge', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(CodingChallenge, {
      propsData: {
        optionsData: sampleData
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
