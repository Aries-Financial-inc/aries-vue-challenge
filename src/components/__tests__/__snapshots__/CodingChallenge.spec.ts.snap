// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodingChallenge renders correctly 1`] = `
<div>
  <h1>Options Profit Calculator</h1>
  <profitlosschart-stub optionsdata="[object Object],[object Object],[object Object],[object Object]" class="Chart"></profitlosschart-stub>
  <profitlosssummary-stub optionsdata="[object Object],[object Object],[object Object],[object Object]" class="Summary"></profitlosssummary-stub>
</div>
`;
